Tests/AWSLambdaRuntimeTests/LambdaLocalServerTests.swift

```diff
@@ -0,0 +1,95 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftAWSLambdaRuntime open source project
+//
+// Copyright (c) 2025 Apple Inc. and the SwiftAWSLambdaRuntime project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Logging
+import NIOCore
+import NIOPosix
+import Testing
+
+@testable import AWSLambdaRuntime
+
+@Suite("Lambda Local Server Tests")
+struct LambdaLocalServerTests {
+
+    @Test("Local server respects LOCAL_LAMBDA_PORT environment variable")
+    func testLocalServerCustomPort() async throws {
+        let customPort = 8080
+
+        // Set environment variable
+        setenv("LOCAL_LAMBDA_PORT", "\(customPort)", 1)
+        defer { unsetenv("LOCAL_LAMBDA_PORT") }
```

Comment:
Given these tests are running in parallel, there is a chance the LOCAL_LAMBDA_PORT environment var setting will affect other tests.

Reply:
@adam-fowler During the tests, we let the MockServer bind on any available port (port = 0) and share its actual port number with the runtime client. There is no risk of conflict. 

See https://github.com/swift-server/swift-aws-lambda-runtime/blob/ec28c9669684bbbe22e447b24c201add4480e8b0/Tests/AWSLambdaRuntimeTests/MockLambdaServer.swift#L28

I kept the `.serialized` only on the new test in case someone adds another test to this suite later on.